js/hw_3_js_react_1/src/Components/GameBoard.js

```diff
@@ -0,0 +1,181 @@
+import React, {useEffect,useState} from 'react'
+import "../styles/GameBoard.css";
+import Timer from './Timer';
+import Score from './Score';
+import { initialTimer,Up,Down,Left,Right,Space,initialSnake,extraFoodTime,board} from '../constants';
+import { getClassName,
+    generateExtraFood,
+    generateFood,
+    getIsCollapsed,
+    getIsExtraFoodEaten,
+    getIsFoodEaten,
+    getIsHitBorder,
+    getSnakeBodyExtraTail,
+    growSnake } from '../utils';
+
+
+const GameBoard = () => {
+  const [score,setScore] = useState(0);
+  const [timer,setTimer] = useState(initialTimer);
+  const [gameLost,setGameLost] = useState(false);
+  const [food,setFood] = useState(generateFood(initialSnake,[]));
+  const [pause,setPause] = useState(false);
+  const [direction,setDirection] = useState('Up');
+  const [snake,setSnake] = useState(initialSnake);
+  const [extraFood,setExtraFood] = useState(undefined);
+
+  useEffect(() => { 
+    document.addEventListener('keydown',handlePressKey);
+    return () => {
+     document.removeEventListener('keydown',handlePressKey)
+    }
+  
+
+  })
+
+
+useEffect(() => {
+  if(pause){
+    return;
+  }
+    const prevSnakeBody = [...snake];
```

Comment:
You don't have to copy it here
```suggestion
    const prevSnakeBody = snake;
```